src/base/StrBuf.cpp

```diff
@@ -0,0 +1,49 @@
+#include "StrBuf.hpp"
+#include <memory.h>
+
+//--------------------------------------------------------------------------------
+StrBuf::StrBuf(size_t capacity, char* buffer)
+  : _buf(buffer)
+  , _dim(capacity)
+  , _len(0)
+{
+  if(buffer)
+    _buf[0] = '\0';
+}
+
+//--------------------------------------------------------------------------------
+void StrBuf::set(size_t capacity, char* buffer)
+{// associate with a new buffer[capacity]
```

Comment:
The comment looks like a description of the function. 
please move it to its function comment section. You can refer to any existing function for references.